### ISSUE TITLE
VST-32 does not compile with Coq 8.14.0 (64 bit is fine)

### ISSUE THREAD

**Issue by MSoegtropIMC** [PrincetonUniversity/VST#526]
Dear VST Team,

In this opam PR:

(https://github.com/coq/opam-coq-archive/pull/1871#issuecomment-947055622)

I tried to enable VST 2.8 for Coq 8.14.0. It works fine for 64 bit, but it fails for 32 bit. See the error message shown above.

VST-32 2.8 compiles fine with Coq 8.13.2.

Best regards,

Michael

**Comment by andrew-appel** [comment]
@palmskog @txyyss @mansky1 
OK, so this is the problem:  Some proof in progs/ghosts.v has different hypothesis names above the line in Coq 8.14 than in Coq 8.13, so the reference to H0 fails in 8.14.  This only shows up in 32-bit mode because, in VST 2.8, ghosts.v is not used in 64-bit mode.  

In the VST master branch, progs/ghosts.v is moved to concurrency/ghosts.v and is supposed to work in either 32-bit or 64-bit mode.)

Therefore, one way to diagnose and fix this this would be to build the VST master branch (in either 64-bit or 32-bit mode) in Coq 8.14 up to concurrency/ghosts.vo, and see which proof is breaking, and fix it

**Comment by palmskog** [comment]
I was able to reproduce the problem, here is the raw diff which pinpoints it and fixes it in the most compatible way I could think of:
```diff
--- a/progs/ghosts.v      2021-05-29 14:09:20.000000000 +0200
+++ b/progs/ghosts.v   2021-10-20 16:43:28.021280621 +0200
@@ -1100,7 +1100,7 @@
 Proof.
   split; [apply all_disjoint_rev1|].
   intros ?%all_disjoint_rev1.
-  rewrite rev_involutive in H0; auto.
+  rewrite rev_involutive in *; auto.
 Qed.
 
 Lemma  maps_add_rev : forall l, all_compatible l -> maps_add (rev l) = maps_add l.
```
However, there may be other similar build issues, so it will take a bit more time to validate and package.

**Comment by palmskog** [comment]
It was confirmed in coq/opam-coq-archive#1877 that the above patch fixes the problem with 32-bit VST 2.8 and Coq 8.14.0. If the "fix" is not relevant for VST `master`, I think there is nothing more to do and you can close this issue.
